src/library.js

```diff
@@ -1163,6 +1161,20 @@ addToLibrary({
         }
       }
 
+      // time zone
+      if ((value = getMatch('z'))) {
+        // GMT offset as either 'Z' or +-HH:MM or +-HH or +-HHMM
+        if (value.toLowerCase() === 'z'){
+          date.gmtoff = 0;
+        } else {          
+          var match = value.match(/^(?<hours>(?:\-|\+)\d\d):?(?<mins>\d\d)?/);
```

Comment:
Unfortunately, we still target ES5. We use some ES6 features where possible by transpiling them via Closure Compiler, but it looks like named capture groups are not one of such features.

![image](https://github.com/emscripten-core/emscripten/assets/557590/27189941-7a90-491a-b5f9-9431e100cb21)

I'm afraid this will need to be converted to regular old capture group.

Reply:
Have done. Also added a test for regex characters in the string which passes which is nice..